src/Icicle/Source/PrettyAnnot.hs

```diff
@@ -1,74 +1,133 @@
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE PatternGuards     #-}
 module Icicle.Source.PrettyAnnot (
     PrettyAnnot(..)
   ) where
 
+import                  Icicle.Source.Type
 import                  Icicle.Source.Query
 import                  Icicle.Internal.Pretty
 
 import                  P
 
-import                  Data.List (intersperse)
-
 newtype PrettyAnnot q
  = PrettyAnnot
  { getPrettyAnnot :: q }
 
-instance (Pretty a, Pretty n) => Pretty (PrettyAnnot (Exp a n)) where
+instance (Pretty a, Pretty n) => Pretty (PrettyAnnot (Exp (Annot a n) n)) where
```

Comment:
They could only really be added if we only had pretty on the annotated queries (which might not actually be that bad) or we had overlapping instances.
It really easy to not print them, just use display instead of displaywithannotation.